### PR TITLE
Use Guava Compatible immediate executor service

### PR DESCRIPTION
**Pull request by drcrallen**
As a follow on to https://github.com/apache/incubator-druid/pull/5413 this PR copies the DirectExecutorService implementation from more recent Guava versions (18 and later). 

I *think* this is the last change to make the druid code base work for a HUGE swath of Guava versions.

### CODE REVIEW

**Comment by jihoonson** [comment]
Is it worth to make Druid compatible with various Guava version? Why don't we implement our own direct executor (as you did in #5413) or simply copy the latest Guava implementation?

**Comment by b-slim** [comment]
i Agree with @jihoonson i think moving away from Guava by copying pasting code or doing our own stuff is far more worth it than trying to support multiple guava versions.

**Comment by drcrallen** [comment]
@jihoonson I tried the "our own implementation" route, and there was too much nuance that I didn't want to deal with. Just fetching the guava stuff correctly was much easier, and puts the maintenance of proper implementation on the Guava libs instead of us.

**Comment by gianm** [comment]
+1 for c/p'ing the latest Guava implementation as-is, like we did with Closer and splitToList. At least then we'll know what we're dealing with.

**Comment by drcrallen** [comment]
Copied over the Direct Executor Service. Luckily it isn't doing any package private mumbo jumbo, so the copy was pretty straight forward

**Comment by drcrallen** [comment]
I can't see why teamcity is failing

**Comment by gianm** [comment]
@drcrallen It looks like there's a reference to CallerRunsPolicy in a javadoc, a class that was not copied over. (You should be able to see this stuff by registering for a TeamCity acct.) Probably the right thing to do is edit the javadoc.

image

**Comment by drcrallen** [comment]
oh! login as guest works well now.

File: processing/src/test/java/org/apache/druid/query/groupby/GroupByQueryRunnerTest.java [hunk 1]
```diff
@@ -29,7 +29,6 @@
 import com.google.common.collect.Lists;
 import com.google.common.collect.Ordering;
 import com.google.common.collect.Sets;
-import com.google.common.util.concurrent.MoreExecutors;
```
Comment (gianm) [inline processing/src/test/java/org/apache/druid/query/groupby/GroupByQueryRunnerTest.java]: Were the changes in this file all autoformatter changes? They are hard to follow so I'd rather ask than try to figure that out.

Comment (drcrallen) [reply processing/src/test/java/org/apache/druid/query/groupby/GroupByQueryRunnerTest.java]: Either auto-formatter or `MoreExecutors.sameThreadExecutor()` --> `Execs.directExecutor()`, yes

**Comment by drcrallen** [comment]
Ugh, if the import is there, failure because unused import (even though it is used in javadoc). If the import is not there it gets failure in teamcity due to unknown reference.

**Comment by gianm** [comment]
I have never really found a good way around that particular weirdness with imports + javadocs. I usually deal with it by using fully qualified class names in javadocs (rather than import + short name). Or by avoiding `@link`.

**Comment by b-slim** [comment]
@drcrallen thanks for doing this!

**Comment by drcrallen** [comment]
For future folks: https://github.com/apache/incubator-druid/pull/6948 adds more compat up to 27.0.1
